### PR TITLE
Show product availability

### PR DESCRIPTION
**Pull request by vsseixaso**

### CODE REVIEW

File: src/main/webapp/app/core/main/controllers/productController.js [hunk 1]
```diff
@@ -20,10 +20,49 @@ app.controller("ProductCtrl", function ($scope, $uibModal, $http, toastr,$locati
...
+                    checksForAllBatches(product);
+                    updateProduct(product);
+                }, function errorCallback(error) {
+                });
+        });
+    };
+
+    var checksForAllBatches = function (product) {
+        _.forEach(productCtrl.productsBatches[product.barCode], function (batch) {
+            if (batch.expirationDate < todaysDate() && product.quantity > 0)
```
Comment (raonismaneoto) [inline src/main/webapp/app/core/main/controllers/productController.js]: && !_.Empty(product.quantity)

File: src/main/webapp/app/core/main/controllers/productController.js [hunk 2]
```diff
@@ -20,10 +20,49 @@ app.controller("ProductCtrl", function ($scope, $uibModal, $http, toastr,$locati
...
+                product.quantity -= batch.numberOfItems;        
+        });
+        if (product.quantity <= 0) product.statusCode = 2;
```
Comment (raonismaneoto) [inline src/main/webapp/app/core/main/controllers/productController.js]: <=? Isn't dangerous? Use lodash functions...

File: src/main/webapp/app/core/main/controllers/productController.js [hunk 3]
```diff
@@ -20,10 +20,49 @@ app.controller("ProductCtrl", function ($scope, $uibModal, $http, toastr,$locati
...
+    };
+
+    var updateProduct = function (product) {
+        ProductService.updateProduct(product.barCode, product)
+            .then(function successCallback(response) {
```
Comment (raonismaneoto) [inline src/main/webapp/app/core/main/controllers/productController.js]: then(function success(response)) {
product = response.data
}

File: src/main/webapp/app/core/main/controllers/productController.js [hunk 4]
```diff
@@ -20,10 +20,49 @@ app.controller("ProductCtrl", function ($scope, $uibModal, $http, toastr,$locati
...
+            }, function errorCallback(error) {
+            });
+    };
+
+    var todaysDate = function () {
+        var date = new Date();
+        var day = date.getDate();
+        var month = date.getMonth() + 1;
```
Comment (raonismaneoto) [inline src/main/webapp/app/core/main/controllers/productController.js]: Magic number 1, extract to a well named constant.

File: src/main/webapp/app/core/main/controllers/productController.js (hunk 2 above)
Comment (raonismaneoto) [inline src/main/webapp/app/core/main/controllers/productController.js]: there is a constant in the constants.js that has all the product status properties. So you can't set to 2, set it to constant.AVAILABLE.value

File: src/main/webapp/app/core/main/controllers/productController.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
-app.controller("ProductCtrl", function ($scope, $uibModal, $http, toastr,$location, ProductService, AuthService, ProductStatus) {
+app.controller("ProductCtrl", function ($scope, $uibModal, $http, toastr,$location, ProductService, AuthService, ProductStatus, dateFilter) {
```
Comment (raonismaneoto) [inline src/main/webapp/app/core/main/controllers/productController.js]: are you using this dateFilter in anywhere?

Comment (vsseixaso) [reply src/main/webapp/app/core/main/controllers/productController.js]: no more, i will remove

**Pull request merged**